masq_lib/src/percentage.rs

```diff
@@ -0,0 +1,257 @@
+// Copyright (c) 2024, MASQ (https://masq.ai) and/or its affiliates. All rights reserved.
+
+use num::integer::mod_floor;
+use num::CheckedAdd;
+use num::CheckedSub;
+use num::{CheckedDiv, CheckedMul, Integer};
+use std::any::type_name;
+use std::fmt::Debug;
+use std::ops::Div;
+use std::ops::Mul;
+
+// It's designed for a storage of values from 0 to 100, after which it can be used to compute
+// the corresponding portion of many integer types. It should also take care of the least significant
+// digit in order to diminish the effect of a precision loss genuinly implied by this kind of math
+// operations done on integers.
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct Percentage {
+    per_cent: u8,
+}
+
+impl Percentage {
+    pub fn new(num: u8) -> Self {
```

Comment:
Maybe you'd like the approach of `try_from` here. https://doc.rust-lang.org/std/convert/trait.TryFrom.html